google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DoManagedVmDeployment.java

```diff
@@ -73,30 +94,40 @@ public void run() {
           loggingHandler,
           "Created temporary staging directory: " + stagingDirectory.getAbsolutePath());
     } catch (IOException e) {
-      throw new RuntimeException(e);
+      logger.error(e);
+      callback.errorOccurred(
+          GctBundle.message("appengine.deployment.error.creating.staging.directory"));
+      return;
     }
 
-    File stagedArtifactPath =
-        copyFile(stagingDirectory, "target" + artifactType, deploymentArtifactPath);
-    stagedArtifactPath.setReadable(true /* readable */, false /* ownerOnly */);
+    try {
+      File stagedArtifactPath =
+          copyFile(stagingDirectory, "target" + artifactType, deploymentArtifactPath);
+      stagedArtifactPath.setReadable(true /* readable */, false /* ownerOnly */);
 
-    copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
-    copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
+      copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
+      copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
+    } catch (IOException e) {
+      logger.error(e);
+      callback.errorOccurred(GctBundle.message("appengine.deployment.error.during.staging"));
```

Comment:
looks like there are a few more missing returns in the class. I assume that after each callback.errorOccurred .. we want to return immediately.